pandas/core/indexes/base.py

```diff
@@ -4426,6 +4427,15 @@ def join(
         (Index([1, 2, 3, 4, 5, 6], dtype='int64'),
         array([ 0,  1,  2, -1, -1, -1]), array([-1, -1, -1,  0,  1,  2]))
         """
+        if not isinstance(other, Index):
+            warnings.warn(
+                f"Passing a Series to {type(self).__name__}.join is deprecated "
```

Comment:
This message will only be correct for a Series other, but the condition is only for non-Index.  can you either make the message more general or condition more specific